### DISCUSSION TITLE
Add-PnPFile

### DISCUSSION THREAD

**Discussion by davidbmott** [pnp/powershell#2923] [General] [answered]
Are there parameters to force an overwrite of a file?

I am running the following code... I am placing thousands of documents in a library, where sometimes the file already exists, and needs to be updated to a newer version of the file.  The files are PDF's. I want to ensure that every time I Add-PnPfile, any existing file is overwritten.  Can you help?
`Add-PnPFile -Path $filePath -Folder $folder -Values @{"COMPLEX_x0020_NAME" = $ReportingAreaTag;  "Folder" = $DestinationFolder; "MONTH" = $MonthYear; "Title" = $FileName; "ComplexNameMetadataTag" = "$($Tag.Id)"; } -ErrorAction Stop`

**Comment by AndersRask** [comment, marked as answer]
I belive this is default?

I guess it would depend on the version settings and if you require check out/content approval (where you have settings on Add-PnPFile to handle these scenarios), but adding the same file should overwrite existing files just fine.

**Reply by davidbmott** [reply]
Thank You for the quick reply 👍
